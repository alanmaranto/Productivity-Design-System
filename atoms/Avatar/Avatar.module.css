.avatar {
  border-radius: var(--border-radius-full);
  border: 1px solid tomato;
  display: inline-block;
}
