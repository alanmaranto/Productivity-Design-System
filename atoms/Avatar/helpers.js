const avatarSize = {
  sm: 15,
  md: 26,
  lg: 30,
  xl: 40,
};

export const mapSize = (size) => avatarSize[size];
