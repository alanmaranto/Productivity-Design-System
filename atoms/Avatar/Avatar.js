import React from "react";
import PropTypes from "prop-types";
import Picture from "../Picture";
import { mapSize } from "./helpers";
import styles from "./Avatar.module.css";
import { options } from "./constants";
import withStyles from "../../hoc/withStyles";

const Avatar = ({ src, size, styles }) => {
  return (
    <div className={styles("avatar")}>
      <Picture
        src={src}
        width={mapSize(size)}
        height={mapSize(size)}
        isRounded
      />
    </div>
  );
};

Avatar.defaultProps = {
  size: "md",
};

Avatar.propTypes = {
  size: PropTypes.oneOf(options.sizes).isRequired,
  src: PropTypes.string.isRequired,
  styles: PropTypes.func.isRequired,
};

export default withStyles(styles)(Avatar);
