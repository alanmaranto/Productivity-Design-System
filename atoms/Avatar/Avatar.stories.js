import Avatar, { options } from ".";
import {
  getOptionsArgTypes,
  getTemplate,
  getListTemplate,
} from "../../helpers/storybook";

export default {
  title: "Atoms/Avatar",
  component: Avatar,
  args: {
    __sb: { fd: "row " },
    src: "https://placeimg.com/200/200/any",
  },
  argTypes: {
    size: getOptionsArgTypes(options.sizes),
  },
};

const Template = getTemplate(Avatar);
const ListTemplate = getListTemplate(Avatar);

export const Default = Template.bind({});

export const Sizes = ListTemplate.bind({});
Sizes.args = {
  items: options.sizes.map((size) => ({ size })),
};
