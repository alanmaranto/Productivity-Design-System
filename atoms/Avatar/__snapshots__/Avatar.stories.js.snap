// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Avatar Default 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="avatar"
  >
    <picture
      className="picture is-rounded"
    >
      <img
        src="https://placeimg.com/200/200/any"
        style={
          Object {
            "height": 44,
            "maxWidth": "44px",
          }
        }
      />
    </picture>
  </div>
</div>
`;

exports[`Storyshots Atoms/Avatar Sizes 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="avatar"
  >
    <picture
      className="picture is-rounded"
    >
      <img
        src="https://placeimg.com/200/200/any"
        style={
          Object {
            "height": 22,
            "maxWidth": "22px",
          }
        }
      />
    </picture>
  </div>
  <div
    className="avatar"
  >
    <picture
      className="picture is-rounded"
    >
      <img
        src="https://placeimg.com/200/200/any"
        style={
          Object {
            "height": 44,
            "maxWidth": "44px",
          }
        }
      />
    </picture>
  </div>
  <div
    className="avatar"
  >
    <picture
      className="picture is-rounded"
    >
      <img
        src="https://placeimg.com/200/200/any"
        style={
          Object {
            "height": 66,
            "maxWidth": "66px",
          }
        }
      />
    </picture>
  </div>
</div>
`;
